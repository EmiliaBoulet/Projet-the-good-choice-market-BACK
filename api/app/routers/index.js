//* Import de Router de express
import Router from "express";
const indexRouter = Router();

import userRouter from "./userRouter.js";
indexRouter.use("/api", userRouter);

import tvaRouter from "./tvaRouter.js";
indexRouter.use("/api", tvaRouter);

import productRouter from "./productRouter.js";
indexRouter.use("/api", productRouter);

import companyRouter from "./companyRouter.js";
indexRouter.use("/api", companyRouter);

import categoryRouter from "./categoryRouter.js";
indexRouter.use("/api", categoryRouter);

import brandRouter from "./brandRouter.js";
indexRouter.use("/api", brandRouter);

import imageRouter from "./imageRouter.js";
indexRouter.use("/api", imageRouter);

import errorRouter from "./errorRouter.js";
indexRouter.use("*", errorRouter);

export default indexRouter;
