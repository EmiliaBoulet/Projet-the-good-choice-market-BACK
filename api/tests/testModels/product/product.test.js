import request from "supertest";
import app from "../../appTest.js";

beforeAll(async () => {
    await request(app).post("/api/user/login").send({
        email: "pasparla@wanadoo.fr",
        password: "12345Aa!"
    });
});

describe("GET ALL /product", () => {
    describe("Check status of response", () => {
        it("response with json", async () => {
            const response = await request(app).get("/api/product");
            expect(response.type).toBe("application/json");
            expect(response.type).not.toBe("application/xml");
        });
        it("response status code 200", async () => {
            const response = await request(app).get("/api/product");
            expect(response.statusCode).toBe(200);
            expect(response.statusCode).not.toBe(404);
            expect(response.statusCode).not.toBe(401);
            expect(response.statusCode).not.toBe(500);
        });
    });

    describe("Check contains of response", () => {
        it("response body contains > 0", async () => {
            const response = await request(app).get("/api/product");
            expect(response.body.length).toBeGreaterThan(0);
        });
    });
});

describe("GET ONE /product", () => {
    const productId = 100;

    describe("Check status of response", () => {
        it("response with json", async () => {
            const response = await request(app).get(`/api/product/${productId}`);
            expect(response.type).toBe("application/json");
        });
        it("response status code 200", async () => {
            const response = await request(app).get(`/api/product/${productId}`);
            expect(response.statusCode).toBe(200);
        });
    });

    describe("Check contains of response", () => {
        it("response body contains > 0", async () => {
            const response = await request(app).get(`/api/product/${productId}`);
            expect(response.body).toMatchSnapshot({
                description: expect.not.stringMatching("UwU"),
                id: expect.any(Number),
                name: expect.stringMatching(/[a-zA-Z]/),
            });
            expect(response.body).not.toContain({"error": "Il n'y a pas de produit correspondant!"});

        });
    });
});

describe("POST ONE /product", () => {
    const productId = 9;

    describe("Check status of response", () => {
        it("response with json", async () => {
            const response = await request(app).post(`/api/product/${productId}/user`);
            expect(response.type).toBe("application/json");
            expect(response.type).not.toBe("application/xml");
        });

        it("response status code 200", async () => {
            const response = await request(app).get("/api/product/9/user");
            expect(response.statusCode).toBe(404);
            expect(response.statusCode).not.toBe(200);
            expect(response.statusCode).not.toBe(401);
            expect(response.statusCode).not.toBe(500);
        });
    });
});
