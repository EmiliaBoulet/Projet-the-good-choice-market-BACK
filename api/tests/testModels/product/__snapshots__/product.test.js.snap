// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET ONE /product Check contains of response response body contains property 1`] = `
Object {
  "barcode": null,
  "brand_id": 1,
  "brand_name": "Les gourmands verts",
  "category_name": "loisirs",
  "created_at": "2022-08-05T14:25:17.071Z",
  "depth": null,
  "description": StringNotMatching /UwU/,
  "expiration_date": null,
  "height": null,
  "id": Any<Number>,
  "image": "https://res.cloudinary.com/drjw8g2yn/image/upload/v1658906849/visuels%20produits/Loisirs/Les%20gourmands%20Verts/coursCuisine_miniature_wsv61x.jpg",
  "location": "Toulouse",
  "name": StringMatching /\\[a-zA-Z\\]/,
  "price_ht": "99.00",
  "price_kg": null,
  "price_promo": null,
  "price_ttc": "99.00",
  "product_id": 1,
  "product_name": StringMatching /\\[a-zA-Z\\]/,
  "size": null,
  "sub_category_id": 13,
  "sub_category_name": "Cuisine",
  "tva_id": 5,
  "updated_at": null,
  "weight": null,
  "width": null,
}
`;
